src/main/java/me/voidxwalker/autoreset/mixin/config/CreateWorldScreenMixin.java

```diff
@@ -94,6 +100,27 @@ private void loadAtumConfigurations(CallbackInfo ci) {
             return;
         }
 
+        if (Atum.isRunning()) {
+            SeedProvider seedProvider = Atum.getSeedProvider();
+            Optional<String> seedOpt = seedProvider.getSeed();
+            if (seedOpt.isPresent()) {
+                seed = seedOpt.get();
+            } else {
+                if (MinecraftClient.getInstance().isOnThread()) {
+                    this.shouldOpenWaitingScreen = true;
+                    return;
+                } else {
+                    while (!seedOpt.isPresent()) {
+                        seedProvider.waitForSeed();
+                        seedOpt = seedProvider.getSeed();
+                    }
+                    seed = seedOpt.get();
+                }
+            }
+        } else {
+            seed = Atum.config.seed; // Set seed for config screen
+        }
+
```

Comment:
I think this code can just be moved into `modifyAtumCreateWorldScreen` so we don't need two extra fields, you can make an extra method in `IMoreOptionsDialog` for setting the seed string